syntax = "proto3";

package fennel.proto;

import "schema.proto";
import "status.proto";

enum AggregateType {
  SUM = 0;
  AVG = 1;
  COUNT = 2;
  MIN = 3;
  MAX = 4;
  RATE = 5;
  KEY_VALUE = 6;
  TOPK = 7;
  CF = 8;
}

message CreateAggregateRequest {
  string name = 1;
  string mode = 2;
  uint32 version = 3;
  string stream = 4;
  bytes agg_cls = 6;
  string function_source_code = 7;
  AggregateType aggregate_type = 8;
  repeated int32 windows = 9;
  Schema schema = 10;
}
